app/src/main/java/com/amaze/filemanager/ui/dialogs/SmbConnectDialog.java

```diff
@@ -230,7 +237,9 @@ public void afterTextChanged(Editable s) {
           chkSmbAnonymous.setChecked(true);
         }
         ipp = a.getHost();
+        sharep = a.getPath().replaceAll("/", "");
```

Comment:
Why replace all the slashes? Isn't it better to support a custom path instead of only root ones?
Eg. you're adding support for only `/A` or `/B` what if I want to add `/A/B/C` as Share path? You'll end up showing `ABC` as Share name to user. So I suggest just replace the first / occurrence while showing in edit text. 
That said, I think the feature text should be named `Share path` instead of `Share Name`

Would also request another change while you're at it.
Since we're adding one more new field to the dialog, due to increased number of fields it's better as UX standpoint to define which fields are optional and which are mandatory.

So request you to name `Share path (optional)` as the edit text hint, also check other fields as well. Eg. domain and name too can be optional. So you can add optional in their name as well.

Reply:
Good call! That makes sense. Since I was only thinking of the share name originally, I figured there would be a slash before and a slash after because those are added to the share path URL (hence replace all slashes). By renaming the field to allow full paths (and clarify which fields are optional) I should only have to remove the first and last slash and it will give the user more flexibility... in fact, a flexibility that will benefit me as well! 

I will get these changes done today :)

Reply:
Looking in the code (and through testing) it looks like the name is *not* optional, so I only labeled `Domain` and `Share path` as optional. You can now add a path and it will connect automatically to where you want.... brilliant!

Hate to bother again, but if you could just double check that an empty share name still connects properly, that would be great. Sorry that I don't have a non-hidden share and keep putting this burden on you :(